src/ray/core_worker/core_worker.cc

```diff
@@ -1949,38 +1949,49 @@ Status CoreWorker::AllocateReturnObjects(
   RAY_CHECK(object_ids.size() == data_sizes.size());
   return_objects->resize(object_ids.size(), nullptr);
 
+  for (size_t i = 0; i < object_ids.size(); i++) {
+    AllocateReturnObject(object_ids[i], data_sizes[i], metadatas[i],
+                         contained_object_ids[i], return_objects->at(i));
+  }
+
+  return Status::OK();
+}
+
+Status CoreWorker::AllocateReturnObject(const ObjectID &object_id,
+                                        const size_t &data_size,
+                                        const std::shared_ptr<Buffer> &metadata,
+                                        const std::vector<ObjectID> &contained_object_id,
+                                        std::shared_ptr<RayObject> &return_object) {
```

Comment:
```suggestion
                                        std::shared_ptr<RayObject> *return_object) {
```
[C++ style guide](https://google.github.io/styleguide/cppguide.html) recommends refs for input arguments and pointers for output arguments.